litho-core/src/main/java/com/facebook/litho/ComponentTree.java

```diff
@@ -1772,7 +1772,10 @@ private void calculateLayout(
     final PerfEvent layoutEvent =
         logger != null
             ? LogTreePopulator.populatePerfEventFromLogger(
-                mContext, logger, logger.newPerformanceEvent(mContext, EVENT_LAYOUT_CALCULATE))
+                mContext,
+                mContext.getTreeProps() == null ? treeProps : mContext.getTreeProps(),
```

Comment:
According to the nearby usage, `context.getTreeProps` is used as a fallback. Let's utilize the same logic here: just passing `treeProps` in, and falling back in overridden `populatePerfEventFromLogger` method.

Reply:
Good point. updated.